### PR TITLE
Allow building for RISC-V and Web-Assembly

### PR DESCRIPTION
**Pull request by falkTX**
Using simde we can enable RISC-V builds.
Depends on https://github.com/surge-synthesizer/sst-plugininfra/pull/37

### CODE REVIEW

**Comment by baconpaul** [comment]
Since we squash merge that plugininfra hash will change right.? So this will have a bad submodule.

the change is fine once we understand if the plugininfra change is fine but will probably have to change the pr once the submodule is updated

**Comment by falkTX** [comment]
I can push without the submodule update if that helps

**Comment by baconpaul** [comment]
Nah the yull just need to push again, leave it here and when we sort out the submodule just relish this with the right hash. Was just letting you know the change is fine

**Comment by baconpaul** [comment]
Anyway if you want to update this with the correct head has for sst-plugin-infra i'm happy to merge it.

**Comment by falkTX** [comment]
This should be it, I have surge running on the browser now :)

JUCE stuff and mingw fix is not pushed here, as those are a bit separate things

**Comment by baconpaul** [comment]
Interesting! Does it work ok?

**Comment by falkTX** [comment]
> Interesting! Does it work ok?

Let me figure out the resources first, and I will let you experiment..

File: src/common/dsp/vembertech/basic_dsp.cpp [hunk 1]
```diff
@@ -262,7 +262,8 @@ float sine_ss(unsigned int x) // using 24-bit range as [0..2PI] input
 
     return P * (y * abs(y) - y) + y;   // Q * y + P * y * abs(y)   */
 }
-#if !_M_X64 && !defined(ARM_NEON)
+
+#if 0
```
Comment (baconpaul) [inline src/common/dsp/vembertech/basic_dsp.cpp:266]: Do you mind just deleting this entire block rather than if 0ing it?

**Review by baconpaul — COMMENTED** [review]
All good other than the if 0 vs remove. Change that and I’m happy to merge!

File: src/common/dsp/vembertech/basic_dsp.cpp (hunk 1 above)
Comment (falkTX) [reply src/common/dsp/vembertech/basic_dsp.cpp]: done

Comment (baconpaul) [reply src/common/dsp/vembertech/basic_dsp.cpp]: Great
When ci finishes I will merge
Thanks for upstreaming

**Comment by falkTX** [comment]
Just leaving a tease...

image

will continue in the evening, need to handle a few other things for now.
still dont have resources in place properly, or rather the path to them.

**Comment by baconpaul** [comment]
The resources only matter for fx presets and the wavetable and window oscillator 

wavetable and window might crash without them but classic and the filters should work

**Comment by falkTX** [comment]
I figured it out, after lots of prints and trying to figure out wtf was going on: emscripten doesnt follow directory symlinks when packing resources, only file symlinks.

With that working I enabled the full packing, but ended up with 320Mb of resources! way too much..
Looking around it seems most of it comes from the 3rd party wavetables, which take 141Mb.
Can you compress those files with flac? Wavetables are short, so shouldnt take long to decode, and would save near 50% of the space.

Sadly even without the 3rd party wavetables the total resource size still ends up being large, thus browsers wont cache it.
With brotli compression and Firefox, we are just on the limit (50.7Mb > Firefox default 50Mb limit).

Anyhow, SurgeXT on Cardinal is now Live! Feel free to try it at https://cardinal.kx.studio/

**Comment by baconpaul** [comment]
You can just skip the wavetables if you want - writing code to compress them is pretty painful since they are a custom format 

I will probably move to a separate download model in 2.1 since both you and rack seem to flinch at the size 

bur for now just package like basic/generated or something and add instructions to your build notes if the size is a problem.

**Comment by mkruselj** [comment]
I would suggest packing the factory wavetables folder, and I don't know from 3rd party wavetables folder maybe include Venus Theory and TNMG ones. Amounts to 17 MB or so - not too terrible?

**Comment by falkTX** [comment]
It is already above the maximum browser cache size, so I went with "factory" wavetables only.
For reducing the size I have been thinking of creating a font pool where modules retrieve fonts from, instead of the current way where each module has their own copy of fonts they use.

**Comment by baconpaul** [comment]
Yeah I think I’m going to add a 2.1 fewture which can download the extra tables from our GitHub and place them in the user working dir we use for other settings - that should strip our assets down to a couple of wavetables we need to get start led and some svg

**Comment by falkTX** [comment]
Sounds fine for general use.
Do note I have a strict rule for Cardinal where internet access is forbidden, really dont want to deal with libcurl, openssl, certificates and all that stuff... plus phone home, analytics yada yada. The lack of online connection blocks that from the start.

I think the current base wavetables are fine, the 3rd party ones I agree would be best stored elsewhere.

**Comment by baconpaul** [comment]
Yup we’ll presumably whatever I write to download the wavetable programmatically will have to be done manually instead by your users but since I haven’t written it yet it’s not a problem today

**Comment by falkTX** [comment]
Any blocker for merging this? I have a few other changes pending, but those will be on different PRs.

**Comment by baconpaul** [comment]
Merged

**Pull request merged**